lib/site_prism/element_container.rb

```diff
@@ -68,16 +73,17 @@ def section(section_name, *args, &block)
         section_class, find_args = extract_section_options(args, &block)
         build(section_name, *find_args) do
           define_method section_name do |*runtime_args, &runtime_block|
-            section_class.new self, find_first(*merge_args(find_args, runtime_args)), &runtime_block
+            section_element = find_first(*merge_args(find_args, runtime_args))
+            section_class.new self, section_element, &runtime_block
           end
         end
       end
 
-      def sections(section_collection_name, *args, &block)
+      def sections(collection_name, *args, &block)
         section_class, find_args = extract_section_options(args, &block)
-        build(section_collection_name, *find_args) do
-          define_method(section_collection_name) do |*runtime_args, &element_block|
-            raise_if_block(self, section_collection_name.to_s, !element_block.nil?)
+        build(collection_name, *find_args) do
```

Comment:
should we call this either `section_name` or even better just `name` because we know it's a section